tests/test_dataelem.py

```diff
@@ -1219,3 +1222,35 @@ def test_invalid_very_long_value_length(self, vr, value):
             DataElement(0x00410001, vr, value, validation_mode=config.WARN)
         with pytest.raises(ValueError, match=msg):
             DataElement(0x00410001, vr, value, validation_mode=config.RAISE)
+
+
+class TestBufferedDataElement:
```

Comment:
Can you add tests for pickling/unpickling buffered elements?

Reply:
I'd also like to know what happens if you use a buffer and the source is then moved/deleted

Reply:
> Can you add tests for pickling/unpickling buffered elements?

Is this an important use-case? I think we'd have the read the buffer into memory for that to work since a BufferedReader isn't pickle-able.

> I'd also like to know what happens if you use a buffer and the source is then moved/deleted

If its before we read the buffer, then the assertion in "buffer assertions" should raise an error on a closed stream. Otherwise, I think it'd be undefined behavior. I can write tests for these specifically if you'd like.

Reply:
I think its more that I'd like to see if there's anywhere to put some nice exception messages about why it failed.